src/test/scala/akka/persistence/cassandra/query/EventsByTagPubsubSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.cassandra.query
+
+import java.nio.ByteBuffer
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+
+import scala.concurrent.duration._
+import scala.util.Try
+
+import akka.actor.ActorSystem
+import akka.persistence.PersistentRepr
+import akka.persistence.cassandra.CassandraLifecycle
+import akka.persistence.cassandra.journal.CassandraJournalConfig
+import akka.persistence.cassandra.journal.CassandraStatements
+import akka.persistence.cassandra.journal.TimeBucket
+import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
+import akka.persistence.cassandra.testkit.CassandraLauncher
+import akka.persistence.journal.Tagged
+import akka.persistence.journal.WriteEventAdapter
+import akka.persistence.query.EventEnvelope
+import akka.persistence.query.PersistenceQuery
+import akka.persistence.query.scaladsl.CurrentEventsByTagQuery
+import akka.persistence.query.scaladsl.EventsByTagQuery
+import akka.serialization.SerializationExtension
+import akka.stream.ActorMaterializer
+import akka.stream.testkit.TestSubscriber
+import akka.stream.testkit.scaladsl.TestSink
+import akka.testkit.ImplicitSender
+import akka.testkit.TestKit
+import com.datastax.driver.core.utils.UUIDs
+import com.typesafe.config.ConfigFactory
+import org.scalatest.Matchers
+import org.scalatest.WordSpecLike
+
+object EventsByTagPubsubSpec {
+  val today = LocalDate.now(ZoneOffset.UTC)
+
+  val config = ConfigFactory.parseString(s"""
+    akka.actor.provider = "akka.cluster.ClusterActorRefProvider"
+    akka.remote.netty.tcp.port = 4872
+    akka.cluster.seed-nodes = ["akka.tcp://EventsByTagPubsubSpec@127.0.0.1:4872"]
```

Comment:
no hardcoded ports in tests, please. You can join programatically instead of seed-nodes

`Cluster(system).join(Cluster(system).selfAddress)`